examples/keras_rs/distributed_embedding_tf.py

```diff
@@ -0,0 +1,315 @@
+"""
+Title: DistributedEmbedding using TPU SparseCore and TensorFlow
+Author: [Fabien Hertschuh](https://github.com/hertschuh/), [Abheesht Sharma](https://github.com/abheesht17/)
+Date created: 2025/09/02
+Last modified: 2025/09/02
+Description: Rank movies using a two tower model with embeddings on SparseCore.
+Accelerator: TPU
+"""
+
+"""
+## Introduction
+
+In the [basic ranking](/keras_rs/examples/basic_ranking/) tutorial, we showed
+how to build a ranking model for the MovieLens dataset to suggest movies to
+users.
+
+This tutorial implements the same model trained on the same dataset but with the
+use of `keras_rs.layers.DistributedEmbedding`, which makes use of SparseCore on
+TPU. This is the TensorFlow version of the tutorial. It needs to be run on TPU
+v5p or v6e.
+
+Let's begin by installing the necessary libraries. Note that we need
+`tensorflow-tpu` version 2.19. We'll also install `keras-rs`.
+"""
+
+"""shell
+pip install -U -q tensorflow-tpu==2.19.1
+pip install -q keras-rs
+"""
+
+"""
+We're using the PJRT version of the runtime for TensorFlow. We're also enabling
+the MLIR bridge. This requires setting a few flags before importing tensorflow.
+"""
+
+import os
+import libtpu
+
+os.environ["PJRT_DEVICE"] = "TPU"
+os.environ["NEXT_PLUGGABLE_DEVICE_USE_C_API"] = "true"
+os.environ["TF_PLUGGABLE_DEVICE_LIBRARY_PATH"] = libtpu.get_library_path()
+os.environ["TF_XLA_FLAGS"] = (
+    "--tf_mlir_enable_mlir_bridge=true "
+    "--tf_mlir_enable_convert_control_to_data_outputs_pass=true "
+    "--tf_mlir_enable_merge_control_flow_pass=true"
+)
+
+import tensorflow as tf
+
+"""
+We're now ready to import `keras` and `keras-rs`. But we need to set the
+backend to TensorFlow.
+"""
+
+os.environ["KERAS_BACKEND"] = "tensorflow"
+
+import keras
+import keras_rs
+import tensorflow_datasets as tfds
+
+"""
+## Creating a TPUStrategy
+
+To run TensorFlow on TPU, you need to use a `tf.distribute.TPUStrategy` to
+handle the distribution of the model.
+
+Note that the core of the model is replicated across TPU instances. Only the
+embedding tables handled by `DistributedEmbedding` are sharded across the
+SparseCore chips of all the available TPUs.
+"""
+
+resolver = tf.distribute.cluster_resolver.TPUClusterResolver(tpu="local")
```

Comment:
[knowledge] I'm assuming this won't work in the multi-host case, right? Since `tpu = "local"`

Reply:
I can't get my hands on a multi-host v6e right now. We can submit as-is and revisit.